client/qiskit_serverless/core/files.py

```diff
@@ -130,15 +130,37 @@ def provider_download(
                 os.path.join(self._files_url, "provider", "download"),
             )
 
-    def upload(self, file: str, provider: Optional[str] = None) -> Optional[str]:
+    def upload(
+        self, file: str, function: QiskitFunction, provider: Optional[str] = None
+    ) -> Optional[str]:
         """Uploads file."""
         tracer = trace.get_tracer("client.tracer")
         with tracer.start_as_current_span("files.upload"):
             with open(file, "rb") as f:
                 with requests.post(
                     os.path.join(self._files_url, "upload"),
                     files={"file": f},
-                    data={"provider": provider},
+                    data={"provider": provider, "function": function.title},
```

Comment:
Correct me if I'm wrong, @korgan00 but I think these need to be `params` instead of data.